### PR TITLE
fix(507): replace panic with error propagation in business rule task

### PR DESCRIPTION
**Pull request by Kuzna**
- Return errors instead of panicking on invalid inputs
- Add tests covering error paths and edge cases

### CODE REVIEW

File: pkg/dmn/dmn_engine.go [hunk 1]
```diff
@@ -353,7 +353,7 @@ func (engine *ZenDmnEngine) evaluateDecision(
 				}
 			}
 		default:
-			panic(fmt.Sprintf("unsupported TInformationRequirement %+v", requirement))
+			return EvaluatedDecisionResult{}, nil, fmt.Errorf("unsupported TInformationRequirement %+v", requirement)
```
[the remark this answers is not in the document]
Comment (Kuzna) [reply pkg/dmn/dmn_engine.go]: fixed

**Review by alisku — APPROVED** [review]
